source/agora/utils/Test.d

```diff
@@ -630,6 +630,12 @@ public auto genesisSpendable () @safe pure nothrow
     return GenesisBlock.spendable;
 }
 
+/// Get random seed for making new block
+public Hash getTestRandomSeed () @safe nothrow
+{
+    return hashFull(Hash.init);
+}
+
```

Comment:
What is the "random seed" supposed to be ?

Reply:
It is supposed to be the return value of `getRandomSeed` of the `Ledger`. 

```
public Hash getRandomSeed() nothrow @safe
{
    return this.slash_man.getRandomSeed(this.last_block.header.height);
}
```

There is no pre-image that is revealed by gossiping in unit tests so the `getTestRandomSeed` is needed.

Reply:
Isn't that a bug though ? That pre-images are not revealed by gossiping ?

Reply:
> Isn't that a bug though ? That pre-images are not revealed by gossiping ?

No, it's not. It's only used in unit tests without any gossiping.